azure-pipelines.yml

```diff
@@ -474,6 +496,7 @@ stages:
           sqMavenPluginVersionChoice: 'latest'
 
       - task: SonarCloudPublish@1
+        condition: always()
```

Comment:
I have not tested this change

Reply:
I failed the task before and this task was run. But I think it does nothing. I get `[warning]No analyses found in this build! Please check your build configuration.` in the [logs](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=52006&view=logs&j=b8df9373-f738-5ca4-a0c1-e467884f9d9b&t=e3135271-93a8-5023-f1f1-95ca509c238b&l=83). We may revert this change.
```
2022-08-04T13:42:13.1847538Z ##[debug]Evaluating condition for step: 'Code Analysis - Publish QG'
2022-08-04T13:42:13.1850225Z ##[debug]Evaluating: always()
2022-08-04T13:42:13.1850928Z ##[debug]Evaluating always:
2022-08-04T13:42:13.1852821Z ##[debug]=> True
2022-08-04T13:42:13.1853691Z ##[debug]Result: True
2022-08-04T13:42:13.1854391Z ##[section]Starting: Code Analysis - Publish QG
2022-08-04T13:42:13.1978727Z ==============================================================================
2022-08-04T13:42:13.1979393Z Task         : Publish Quality Gate Result
2022-08-04T13:42:13.1979851Z Description  : Publish SonarCloud's Quality Gate result on the Azure Pipelines build result. To be used after the actual analysis.
2022-08-04T13:42:13.1980313Z Version      : 1.11.1
2022-08-04T13:42:13.1980606Z Author       : sonarsource
2022-08-04T13:42:13.1981102Z Help         : Version: 1.11.1. [More Information](https://docs.sonarcloud.io/advanced-setup/ci-based-analysis/sonarscanner-for-azure-devops/)
2022-08-04T13:42:13.1981685Z ==============================================================================
2022-08-04T13:42:13.2249008Z ##[debug]Using node path: C:\Program Files\vsts-agent\externals\node10\bin\node.exe
2022-08-04T13:42:13.3700590Z ##[debug]agent.TempDirectory=C:\ProgramData\vsts-agent\_temp
2022-08-04T13:42:13.3723968Z ##[debug]loading inputs and endpoints
2022-08-04T13:42:13.3741458Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2022-08-04T13:42:13.3788345Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2022-08-04T13:42:13.3791102Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2022-08-04T13:42:13.3793203Z ##[debug]loading INPUT_POLLINGTIMEOUTSEC
2022-08-04T13:42:13.3798204Z ##[debug]loading SECRET_ARTIFACTORY_API_KEY
2022-08-04T13:42:13.3801176Z ##[debug]loading SECRET_ARTIFACTORY_DEPLOY_PASSWORD
2022-08-04T13:42:13.3802985Z ##[debug]loading SECRET_ARTIFACTORY_PRIVATE_READER_PASSWORD
2022-08-04T13:42:13.3804660Z ##[debug]loading SECRET_ARTIFACTORY_QA_READER_PASSWORD
2022-08-04T13:42:13.3806660Z ##[debug]loading SECRET_CHOCO_API_KEY
2022-08-04T13:42:13.3808483Z ##[debug]loading SECRET_GITHUB_TOKEN
2022-08-04T13:42:13.3810477Z ##[debug]loading SECRET_JKSPASSWORD
2022-08-04T13:42:13.3812164Z ##[debug]loading SECRET_PFXPASSWORD
2022-08-04T13:42:13.3814101Z ##[debug]loading SECRET_PFXPASSWORD2021-2023
2022-08-04T13:42:13.3815814Z ##[debug]loading SECRET_PFXSHA1
2022-08-04T13:42:13.3817460Z ##[debug]loading SECRET_PFXSHA1-2021-2023
2022-08-04T13:42:13.3819263Z ##[debug]loading SECRET_PGP_KEYID
2022-08-04T13:42:13.3821241Z ##[debug]loading SECRET_PGP_PASSPHRASE
2022-08-04T13:42:13.3822969Z ##[debug]loading SECRET_SONARCLOUD_IT_PASSWORD
2022-08-04T13:42:13.3824357Z ##[debug]loading SECRET_SONARQUBE_ENDPOINT
2022-08-04T13:42:13.3826583Z ##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
2022-08-04T13:42:13.3828178Z ##[debug]loading SECRET_WHITESOURCE_APIKEY
2022-08-04T13:42:13.3834427Z ##[debug]loaded 21
2022-08-04T13:42:13.3838549Z ##[debug]Agent.ProxyUrl=undefined
2022-08-04T13:42:13.3840179Z ##[debug]Agent.CAInfo=undefined
2022-08-04T13:42:13.3841136Z ##[debug]Agent.ClientCert=undefined
2022-08-04T13:42:13.3841923Z ##[debug]Agent.SkipCertValidation=undefined
2022-08-04T13:42:13.7371500Z ##[debug]SONARQUBE_SCANNER_PARAMS={"sonar.host.url":"https://sonarcloud.io/","sonar.login":***,"sonar.organization":"sonarsource","sonar.pullrequest.key":"5957","sonar.pullrequest.base":"master","sonar.pullrequest.branch":"Martin/PublishTestResults","sonar.pullrequest.provider":"github","sonar.pullrequest.github.repository":"SonarSource/sonar-dotnet","sonar.scanner.metadataFilePath":"C:\\ProgramData\\vsts-agent\\_temp\\sonar\\20220804.45\\065903c2-a87d-20a6-80ac-e1f17427b28e\\report-task.txt"}
2022-08-04T13:42:13.7376525Z ##[debug]SONARQUBE_ENDPOINT=***
2022-08-04T13:42:13.7378927Z ##[debug][SQ] API GET: '/api/metrics/search' with query "{"f":"name","ps":500}"
2022-08-04T13:42:13.8266095Z ##[debug]Response: 200 Body: "{"metrics":[{"id":"289","key":"new_technical_debt","type":"WORK_DUR","name":"Added Technical Debt"},{"id":"386","key":"analysis_from_sonarqube_9_4","type":"BOOL","name":"Analysis From SonarQube 9.4"},{"id":"78","key":"blocker_violations","type":"INT","name":"Blocker Issues"},{"id":"308","key":"bugs","type":"INT","name":"Bugs"},{"id":"4","key":"classes","type":"INT","name":"Classes"},{"id":"306","key":"code_smells","type":"INT","name":"Code Smells"},{"id":"347","key":"cognitive_complexity","type":"INT","name":"Cognitive Complexity"},{"id":"32","key":"comment_lines","type":"INT","name":"Comment Lines"},{"id":"239","key":"comment_lines_data","type":"DATA","name":"comment_lines_data"},{"id":"33","key":"comment_lines_density","type":"PERCENT","name":"Comments (%)"},{"id":"27","key":"class_complexity","type":"FLOAT","name":"Complexity / Class"},{"id":"38","key":"file_complexity","type":"FLOAT","name":"Complexity / File"},{"id":"28","key":"function_complexity","type":"FLOAT","name":"Complexity / Function"},{"id":"272","key":"complexity_in_classes","type":"INT","name":"Complexity in Classes"},{"id":"273","key":"complexity_in_functions","type":"INT","name":"Complexity in Functions"},{"id":"97","key":"branch_coverage","type":"PERCENT","name":"Condition Coverage"},{"id":"178","key":"new_branch_coverage","type":"PERCENT","name":"Condition Coverage on New Code"},{"id":"95","key":"conditions_to_cover","type":"INT","name":"Conditions to Cover"},{"id":"176","key":"new_conditions_to_cover","type":"INT","name":"Conditions to Cover on New Code"},{"id":"277","key":"confirmed_issues","type":"INT","name":"Confirmed Issues"},{"id":"20","key":"coverage","type":"PERCENT","name":"Coverage"},{"id":"172","key":"new_coverage","type":"PERCENT","name":"Coverage on New Code"},{"id":"79","key":"critical_violations","type":"INT","name":"Critical Issues"},{"id":"26","key":"complexity","type":"INT","name":"Cyclomatic Complexity"},{"id":"181","key":"last_commit_date","type":"MILLISEC","name":"Date of Last Commit"},{"id":"280","key":"development_cost","type":"STRING","name":"Development Cost"},{"id":"348","key":"new_development_cost","type":"FLOAT","name":"Development Cost on New Code"},{"id":"37","key":"directories","type":"INT","name":"Directories"},{"id":"51","key":"duplicated_blocks","type":"INT","name":"Duplicated Blocks"},{"id":"339","key":"new_duplicated_blocks","type":"INT","name":"Duplicated Blocks on New Code"},{"id":"52","key":"duplicated_files","type":"INT","name":"Duplicated Files"},{"id":"8","key":"duplicated_lines","type":"INT","name":"Duplicated Lines"},{"id":"34","key":"duplicated_lines_density","type":"PERCENT","name":"Duplicated Lines (%)"},{"id":"340","key":"new_duplicated_lines_density","type":"PERCENT","name":"Duplicated Lines (%) on New Code"},{"id":"338","key":"new_duplicated_lines","type":"INT","name":"Duplicated Lines on New Code"},{"id":"53","key":"duplications_data","type":"DATA","name":"Duplication Details"},{"id":"312","key":"effort_to_reach_maintainability_rating_a","type":"WORK_DUR","name":"Effort to Reach Maintainability Rating A"},{"id":"321","key":"executable_lines_data","type":"DATA","name":"executable_lines_data"},{"id":"274","key":"false_positive_issues","type":"INT","name":"False Positive Issues"},{"id":"154","key":"file_complexity_distribution","type":"DISTRIB","name":"File Distribution / Complexity"},{"id":"36","key":"files","type":"INT","name":"Files"},{"id":"59","key":"function_complexity_distribution","type":"DISTRIB","name":"Function Distribution / Complexity"},{"id":"6","key":"functions","type":"INT","name":"Functions"},{"id":"150","key":"generated_lines","type":"INT","name":"Generated Lines"},{"id":"151","key":"generated_ncloc","type":"INT","name":"Generated Lines of Code"},{"id":"82","key":"info_violations","type":"INT","name":"Info Issues"},{"id":"42","key":"violations","type":"INT","name":"Issues"},{"id":"94","key":"line_coverage","type":"PERCENT","name":"Line Coverage"},{"id":"175","key":"new_line_coverage","type":"PERCENT","name":"Line Coverage on New Code"},{"id":"55","key":"lines","type":"INT","name":"Lines"},{"id":"5","key":"ncloc","type":"INT","name":"Lines of Code"},{"id":"293","key":"ncloc_language_distribution","type":"DATA","name":"Lines of Code Per Language"},{"id":"92","key":"lines_to_cover","type":"INT","name":"Lines to Cover"},{"id":"173","key":"new_lines_to_cover","type":"INT","name":"Lines to Cover on New Code"},{"id":"163","key":"sqale_rating","type":"RATING","name":"Maintainability Rating"},{"id":"341","key":"new_maintainability_rating","type":"RATING","name":"Maintainability Rating on New Code"},{"id":"80","key":"major_violations","type":"INT","name":"Major Issues"},{"id":"81","key":"minor_violations","type":"INT","name":"Minor Issues"},{"id":"238","key":"ncloc_data","type":"DATA","name":"ncloc_data"},{"id":"165","key":"new_blocker_violations","type":"INT","name":"New Blocker Issues"},{"id":"309","key":"new_bugs","type":"INT","name":"New Bugs"},{"id":"307","key":"new_code_smells","type":"INT","name":"New Code Smells"},{"id":"166","key":"new_critical_violations","type":"INT","name":"New Critical Issues"},{"id":"169","key":"new_info_violations","type":"INT","name":"New Info Issues"},{"id":"164","key":"new_violations","type":"INT","name":"New Issues"},{"id":"337","key":"new_lines","type":"INT","name":"New Lines"},{"id":"167","key":"new_major_violations","type":"INT","name":"New Major Issues"},{"id":"168","key":"new_minor_violations","type":"INT","name":"New Minor Issues"},{"id":"351","key":"new_security_hotspots","type":"INT","name":"New Security Hotspots"},{"id":"311","key":"new_vulnerabilities","type":"INT","name":"New Vulnerabilities"},{"id":"275","key":"open_issues","type":"INT","name":"Open Issues"},{"id":"47","key":"quality_profiles","type":"DATA","name":"Profiles"},{"id":"240","key":"projects","type":"INT","name":"Project branches"},{"id":"58","key":"public_api","type":"INT","name":"Public API"},{"id":"61","key":"public_documented_api_density","type":"PERCENT","name":"Public Documented API (%)"},{"id":"62","key":"public_undocumented_api","type":"INT","name":"Public Undocumented API"},{"id":"294","key":"quality_gate_details","type":"DATA","name":"Quality Gate Details"},{"id":"48","key":"alert_status","type":"LEVEL","name":"Quality Gate Status"},{"id":"315","key":"reliability_rating","type":"RATING","name":"Reliability Rating"},{"id":"342","key":"new_reliability_rating","type":"RATING","name":"Reliability Rating on New Code"},{"id":"313","key":"reliability_remediation_effort","type":"WORK_DUR","name":"Reliability Remediation Effort"},{"id":"314","key":"new_reliability_remediation_effort","type":"WORK_DUR","name":"Reliability Remediation Effort on New Code"},{"id":"276","key":"reopened_issues","type":"INT","name":"Reopened Issues"},{"id":"350","key":"security_hotspots","type":"INT","name":"Security Hotspots"},{"id":"354","key":"security_hotspots_reviewed","type":"PERCENT","name":"Security Hotspots Reviewed"},{"id":"355","key":"new_security_hotspots_reviewed","type":"PERCENT","name":"Security Hotspots Reviewed on New Code"},{"id":"319","key":"security_rating","type":"RATING","name":"Security Rating"},{"id":"343","key":"new_security_rating","type":"RATING","name":"Security Rating on New Code"},{"id":"317","key":"security_remediation_effort","type":"WORK_DUR","name":"Security Remediation Effort"},{"id":"318","key":"new_security_remediation_effort","type":"WORK_DUR","name":"Security Remediation Effort on New Code"},{"id":"352","key":"security_review_rating","type":"RATING","name":"Security Review Rating"},{"id":"353","key":"new_security_review_rating","type":"RATING","name":"Security Review Rating on New Code"},{"id":"356","key":"security_hotspots_reviewed_status","type":"INT","name":"Security Review Reviewed Status"},{"id":"358","key":"new_security_hotspots_reviewed_status","type":"INT","name":"Security Review Reviewed Status on New Code"},{"id":"357","key":"security_hotspots_to_review_status","type":"INT","name":"Security Review To Review Status"},{"id":"359","key":"new_security_hotspots_to_review_status","type":"INT","name":"Security Review To Review Status on New Code"},{"id":"11","key":"skipped_tests","type":"INT","name":"Skipped Unit Tests"},{"id":"60","key":"statements","type":"INT","name":"Statements"},{"id":"162","key":"sqale_index","type":"WORK_DUR","name":"Technical Debt"},{"id":"297","key":"sqale_debt_ratio","type":"PERCENT","name":"Technical Debt Ratio"},{"id":"304","key":"new_sqale_debt_ratio","type":"PERCENT","name":"Technical Debt Ratio on New Code"},{"id":"96","key":"uncovered_conditions","type":"INT","name":"Uncovered Conditions"},{"id":"177","key":"new_uncovered_conditions","type":"INT","name":"Uncovered Conditions on New Code"},{"id":"93","key":"uncovered_lines","type":"INT","name":"Uncovered Lines"},{"id":"174","key":"new_uncovered_lines","type":"INT","name":"Uncovered Lines on New Code"},{"id":"14","key":"test_execution_time","type":"MILLISEC","name":"Unit Test Duration"},{"id":"10","key":"test_errors","type":"INT","name":"Unit Test Errors"},{"id":"12","key":"test_failures","type":"INT","name":"Unit Test Failures"},{"id":"13","key":"tests","type":"INT","name":"Unit Tests"},{"id":"15","key":"test_success_density","type":"PERCENT","name":"Unit Test Success (%)"},{"id":"310","key":"vulnerabilities","type":"INT","name":"Vulnerabilities"},{"id":"324","key":"wont_fix_issues","type":"INT","name":"Won't Fix Issues"}],"total":112,"p":1,"ps":500}"
2022-08-04T13:42:13.8286888Z ##[debug]pollingTimeoutSec=300
2022-08-04T13:42:13.8287864Z ##[debug][SQ] API GET: '/api/server/version' with query "undefined"
2022-08-04T13:42:13.8598291Z ##[debug]Response: 200 Body: "8.0.0.30661"
2022-08-04T13:42:13.8603865Z ##[debug]Build.BuildNumber=20220804.45
2022-08-04T13:42:13.8604744Z ##[debug]Agent.TempDirectory=C:\ProgramData\vsts-agent\_temp
2022-08-04T13:42:13.8608774Z ##[debug]defaultRoot: 'C:\ProgramData\vsts-agent\_temp'
2022-08-04T13:42:13.8609672Z ##[debug]findOptions.allowBrokenSymbolicLinks: 'false'
2022-08-04T13:42:13.8610737Z ##[debug]findOptions.followSpecifiedSymbolicLink: 'true'
2022-08-04T13:42:13.8611993Z ##[debug]findOptions.followSymbolicLinks: 'true'
2022-08-04T13:42:13.8613145Z ##[debug]findOptions.skipMissingFiles: 'false'
2022-08-04T13:42:13.8613934Z ##[debug]matchOptions.debug: 'false'
2022-08-04T13:42:13.8614694Z ##[debug]matchOptions.nobrace: 'true'
2022-08-04T13:42:13.8615743Z ##[debug]matchOptions.noglobstar: 'false'
2022-08-04T13:42:13.8616485Z ##[debug]matchOptions.dot: 'true'
2022-08-04T13:42:13.8618068Z ##[debug]matchOptions.noext: 'false'
2022-08-04T13:42:13.8618775Z ##[debug]matchOptions.nocase: 'true'
2022-08-04T13:42:13.8619476Z ##[debug]matchOptions.nonull: 'false'
2022-08-04T13:42:13.8620384Z ##[debug]matchOptions.matchBase: 'false'
2022-08-04T13:42:13.8621158Z ##[debug]matchOptions.nocomment: 'false'
2022-08-04T13:42:13.8621884Z ##[debug]matchOptions.nonegate: 'false'
2022-08-04T13:42:13.8622618Z ##[debug]matchOptions.flipNegate: 'false'
2022-08-04T13:42:13.8623366Z ##[debug]pattern: 'sonar\20220804.45\**\report-task.txt'
2022-08-04T13:42:13.8636374Z ##[debug]findPath: 'C:\ProgramData\vsts-agent\_temp\sonar\20220804.45'
2022-08-04T13:42:13.8637392Z ##[debug]statOnly: 'false'
2022-08-04T13:42:13.8638567Z ##[debug]findPath: 'C:\ProgramData\vsts-agent\_temp\sonar\20220804.45'
2022-08-04T13:42:13.8639520Z ##[debug]findOptions.allowBrokenSymbolicLinks: 'false'
2022-08-04T13:42:13.8641437Z ##[debug]findOptions.followSpecifiedSymbolicLink: 'true'
2022-08-04T13:42:13.8642639Z ##[debug]findOptions.followSymbolicLinks: 'true'
2022-08-04T13:42:13.8643436Z ##[debug]findOptions.skipMissingFiles: 'false'
2022-08-04T13:42:13.8644310Z ##[debug]0 results
2022-08-04T13:42:13.8645449Z ##[debug]found 0 paths
2022-08-04T13:42:13.8646454Z ##[debug]applying include pattern
2022-08-04T13:42:13.8647888Z ##[debug]adjustedPattern: 'C:\ProgramData\vsts-agent\_temp\sonar\20220804.45\**\report-task.txt'
2022-08-04T13:42:13.8648763Z ##[debug]0 matches
2022-08-04T13:42:13.8649635Z ##[debug]0 final results
2022-08-04T13:42:13.8650994Z ##[debug][SQ] Searching for sonar\20220804.45\**\report-task.txt - found 0 file(s)
2022-08-04T13:42:13.8652353Z ##[warning]No analyses found in this build! Please check your build configuration.
2022-08-04T13:42:13.8653374Z ##[debug]Processed: ##vso[task.issue type=warning;]No analyses found in this build! Please check your build configuration.
2022-08-04T13:42:13.8654684Z ##[debug]Overall Quality Gate status: ok
2022-08-04T13:42:13.8659113Z ##[debug]System.TeamFoundationCollectionUri=https://dev.azure.com/sonarsource/
2022-08-04T13:42:13.8660398Z ##[debug]System.TeamProjectId=399fb241-ecc7-4802-8697-dcdd01fbb832
2022-08-04T13:42:13.8661184Z ##[debug]Build.BuildId=52006
2022-08-04T13:42:13.8661947Z ##[debug][{"op":0,"path":"/sonarglobalqualitygate","value":"ok"}]
2022-08-04T13:42:13.8663663Z ##[debug]SYSTEMVSSCONNECTION exists true
2022-08-04T13:42:13.8666965Z ##[debug]SYSTEMVSSCONNECTION exists true
2022-08-04T13:42:13.8683785Z ##[debug]Acquiring a build API object.
2022-08-04T13:42:14.0521320Z ##[debug]Creating a new build property with global Quality Gate Status
2022-08-04T13:42:14.1754673Z ##[debug]build.artifactStagingDirectory=C:\ProgramData\vsts-agent\1\a
2022-08-04T13:42:14.1767225Z ##[debug][SQ] Summary saved at: C:\ProgramData\vsts-agent\1\a\.sqAnalysis\SonarQubeBuildSummary.md
2022-08-04T13:42:14.1768588Z ##[debug][SQ] Uploading build summary from C:\ProgramData\vsts-agent\1\a\.sqAnalysis\SonarQubeBuildSummary.md
2022-08-04T13:42:14.1779849Z ##[debug]Processed: ##vso[task.addattachment type=Distributedtask.Core.Summary;name=SonarCloud Analysis Report;]C:\ProgramData\vsts-agent\1\a\.sqAnalysis\SonarQubeBuildSummary.md
2022-08-04T13:42:14.1866423Z ##[section]Finishing: Code Analysis - Publish QG
```

Reply:
Let's revert it. We have SQ as required step so we should be fine without it